### PR TITLE
Wait for pod status at tests

### PR DESCRIPTION
**Pull request by qinqon**
Also print pod logs at AfterEach

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by qinqon** [comment]
ci test please

File: tests/report_status_test.go [hunk 1]
```diff
@@ -11,10 +15,78 @@ import (
...
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == corev1.PodReady &&
+			condition.Status == corev1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func WaitPodsCleanup() {
```
Comment (phoracek) [inline tests/report_status_test.go]: this should return an error, so we can loudly fail if cleanup fails

File: tests/report_status_test.go [hunk 2]
```diff
@@ -11,10 +15,78 @@ import (
...
+	wait.Poll(10, 30*time.Second, func() (bool, error) {
+		podsClient := k8sClientset.CoreV1().Pods(*nmstateNs)
```
Comment (phoracek) [inline tests/report_status_test.go]: can we define this globally, nmstatePodsClient

File: tests/report_status_test.go [hunk 3]
```diff
@@ -11,10 +15,78 @@ import (
...
+		return true, nil
+	})
+}
+
+func GetPodsLogs() string {
+	podLogOpts := corev1.PodLogOptions{}
+	podsClient := k8sClientset.CoreV1().Pods(*nmstateNs)
+	pods, err := podsClient.List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Sprintf("error listing pods: %s\n", err)
```
Comment (phoracek) [inline tests/report_status_test.go]: we should return string, error, so it is easy to check whether we failed or nor

File: tests/report_status_test.go (hunk 1 above)
Comment (qinqon) [reply tests/report_status_test.go]: Done

File: tests/report_status_test.go (hunk 2 above)
Comment (qinqon) [reply tests/report_status_test.go]: Done

File: tests/report_status_test.go (hunk 3 above)
Comment (qinqon) [reply tests/report_status_test.go]: Done only the first error, the other ones just make the loop skip the failed pod and continue with others.

**Comment by qinqon** [comment]
test ci please

**Pull request merged**